src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/SlingInitialContentBundleEntryMetaDataCollector.java

```diff
@@ -106,10 +132,54 @@ Set<SlingInitialContentBundleEntryMetaData> collectFromContextAndWriteTmpFiles()
                 }
             }
         }
-
+        
         return collectedSlingInitialContentBundleEntries;
     }
+    
+    Set<SlingInitialContentBundleEntryMetaData> getDefaultContentXmlEntries() throws UnsupportedEncodingException {
+        
+        createDefaultContentXMLFiles();
+        return defaultContentXmlEntries;
+        
+    }
+
+    private void createDefaultContentXMLFiles() throws UnsupportedEncodingException {
+        
+        for(String repositoryPath: collectedSlingInitialContentBundleEntries.stream()
+                .map(SlingInitialContentBundleEntryMetaData::getRepositoryPath)
+                .collect(Collectors.toList())){
+            addDefaultSlingContentXMLAndParents(repositoryPath);
+        }
+    }
+
+    private void addDefaultSlingContentXMLAndParents(String repositoryPath) throws UnsupportedEncodingException {
```

Comment:
UnsupportedEncodingException is never thrown (according to my IDE). please remove it.

Reply:
add @NotNull annotation to the repositoryPath param